tests/io/test_bedgraphio.py

```diff
@@ -0,0 +1,140 @@
+    intervals at single base-pair resolution. Uses a dict describing \
+    an interval range as input, and compares with expected output. \
+    Scores provides a score for *each* base-pair.
+    """
+    interval = {'chrom': 'chr1', 'start': 1, 'end': 5,
+                'scores': [0.2, 0.5, 0.5, 0.2]}
+    output_df = bedgraphio.expand_interval(interval, score=True)
+    expected_df_dict = {'chrom': ['chr1'] * 4,
+                        'start': [1, 2, 3, 4],
+                        'end': [2, 3, 4, 5],
+                        'score': [0.2, 0.5, 0.5, 0.2]}
+    expected_df = pd.DataFrame(expected_df_dict)
+    assert expected_df.equals(output_df)
+
+
+@pytest.mark.cpu
+def test_contract_interval():
+    """
+    Tests behavior of 'contracting' an interval DataFrame at single-bp \
+    resolution. Contracting entails combining neighboring bases with the \
+    same score into a single DataFrame entry.
+    Note: the last basepair always ends up as its own interval, even if it \
+    has the same score as previous, as reflected in the test's expected DF.
+    """
+    expanded_df_dict = {'chrom': ['chr1'] * 7,
+                        'start': range(1, 8),
+                        'end': range(2, 9),
+                        'score': [0.2, 0.2, 0.5, -0.1, -0.1, 0.5, 0.5]}
+    expanded_df = pd.DataFrame(expanded_df_dict)
+    output_df = bedgraphio.contract_interval(expanded_df, positive=True)
+    output_df.reset_index(drop=True, inplace=True)
+    expected_df_dict = {'chrom': ['chr1'] * 4,
+                        'start': [1, 3, 6, 7],
+                        'end': [3, 4, 7, 8],
+                        'score': [0.2, 0.5, 0.5, 0.5]}
+    expected_df = pd.DataFrame(expected_df_dict)
+    assert expected_df.equals(output_df)
+
+
+@pytest.mark.cpu
+def test_intervals_to_bg_basepair():
+    """
+    Tests behavior of function that maps an intervals DF to bedGraph format. \
+    An arbitrary number of intervals is first expanded to single-bp \
+    resolution, and then these intervals are contracted where possible.
+    Computed DF compared with expected DF.
+    """
+    intervals_df_dict = {'chrom': ['chr1', 'chr1', 'chr2'],
+                         'start': [1, 4, 2],
+                         'end': [4, 7, 5],
+                         'scores': [[0.5, 0.5, 0.1],
+                                    [0.5, 0.5, 0.1],
+                                    [0.1, 0.1, 0.0]]}
+    intervals_df = pd.DataFrame(intervals_df_dict)
+    output_df = bedgraphio.intervals_to_bg(intervals_df, resolution=None)
+    output_df.reset_index(drop=True, inplace=True)
+    expected_df_dict = {'chrom': ['chr1'] * 4 + ['chr2'],
+                        'start': [1, 3, 4, 6, 2],
+                        'end': [3, 4, 6, 7, 4],
+                        'score': [0.5, 0.1, 0.5, 0.1, 0.1]}
+    expected_df = pd.DataFrame(expected_df_dict)
+    pd.testing.assert_frame_equal(expected_df, output_df, check_dtype=False)
```

Comment:
Is there a reason why `equals` API was not used ?

Reply:
`equals` is too stringent, e.g. requiring dtypes to be equal, so it's just more convenient to use `pandas` dataframe-specific function which is specifically designed for this purpose

Reply:
if it's poor style to use `assert_frame_equal()` here and `df.equals()` in the first two tests, I could change all to `assert_frame_equal()`?

Reply:
Agreed, I think we can switch to `assert_frame_equal` for everything - unless @ntadimeti you feel we need the stringent checking?

Reply:
Discussed this with Raj. I would ideally like to keep it stringent, since we expect certain dtypes in the code. If for some reason that's hard to test, I'll look into it. I'm taking over this PR from Raj, as he has other things to get to.